classes/Dealer.js

```diff
@@ -0,0 +1,42 @@
+import { StandardDeck } from './Deck'
+
+class Dealer {
+  constructor(deck) {
+    this.deck = deck
+  }
+
+  shuffle() {
+    const plusMinus = (Math.random() > 0.5) ? 1 : -1
+    const cut = Math.floor(Math.random() * 5) * plusMinus
+    const leftHand = this.deck.cards.splice(0, (this.deck.cards.length / 2) + cut)
+    const rightHand = this.deck.cards.splice(0, this.deck.cards.length)
+    
+    while (leftHand.length && rightHand.length) {
+      let mult = 0.5
+      let hand = Math.random() > mult ? 'left' : 'right'
+      if (hand === 'left') {
+        this.deck.cards.push(leftHand.shift())
```

Comment:
I think it would be very helpful to make util functions which add/remove a card to/from the top/bottom of a deck. It would help readability here, and it seems like there could be a lot of use for that in the future, like for the hole cards or the community cards, etc.

Reply:
Good idea, adding it

Reply:
Actually, I think it should be a function of the dealer, rather than a util function

Reply:
I thought about it too, and I think it should be a function of the Deck class.

A deck is just a collection of cards, not necessarily a 52 cards (which is why I assume is the reason you made the StandardDeck class from the Deck class). It can have many cards, it can have 1 card, it can be empty. So I think Players will have decks of cards too as their hole cards, and community cards would be a deck too.

But the most important thing is that the deck implementation shouldn't be known, and therefore players shouldn't know how to add/remove cards from the deck without calling some operation on it to do that action. I think the addition/removal of cards should be on the deck, not an entity doing it on the deck.